socorro/cron/jobs/fetch_adi_from_hive.py

```diff
@@ -345,13 +352,37 @@ def run(self, date):
             self.config.logger.info(
                 'Wrote %d rows from doing hive query' % rows_written
             )
-            for transaction in transactions:
-                transaction(
-                    self._database_transaction,
-                    raw_adi_logs_pathname,
-                    target_date
-                )
+
+            self.persist_data(transactions, raw_adi_logs_pathname, target_date)
 
         finally:
             if os.path.isfile(raw_adi_logs_pathname):
                 os.remove(raw_adi_logs_pathname)
+
+    def persist_data(self, transactions, raw_adi_logs_pathname, target_date):
+        for transaction in transactions:
+            transaction(
+                self._database_transaction,
+                raw_adi_logs_pathname,
+                target_date
+            )
+
+
+class DryRunFetchADIFromHiveCronApp(FetchADIFromHiveCronApp):
+    """Dry run version of FetchADIFromHiveCronApp that doesn't persist any data
+
+    """
+    app_name = 'dry-run-fetch-adi-from-hive'
+    app_description = 'Dry Run Fetch ADI From Hive App'
+    app_version = '0.1'
+
+    required_config = Namespace()
+
+    # Override the as_backfill_cron_app() decorator
+    _is_backfill_app = False
+
+    def main(self, function=None, once=True):
+        return BaseCronApp.main(self, function, once)
```

Comment:
This explicitly calls the super-super class. The decorator is curious and stomps on the class method and I didn't want to deal with trying to figure out a valid `super` invocation. Instead, I went with this which definitely does what I want it to.